src/us/kbase/workspace/database/provenance/Contributor.java

```diff
@@ -0,0 +1,264 @@
+	 *
+	 * @return the contributor's family name, if present
+	 */
+	public Optional<String> getFamilyName() {
+		return Optional.ofNullable(familyName);
+	}
+
+	/**
+	 * Gets the ID for the contributor.
+	 *
+	 * @return the contributor ID, if present
+	 */
+	public Optional<String> getContributorID() {
+		return Optional.ofNullable(contributorID);
+	}
+
+	/**
+	 * Gets the contributor's affiliations.
+	 *
+	 * @return the contributor's affiliations, if present.
+	 */
+	public List<Organization> getAffiliations() {
+		return Common.getList(affiliations);
+	}
+
+	/**
+	 * Gets the roles that the contributor filled.
+	 *
+	 * @return the contributor roles, if present.
+	 */
+	public List<ContributorRole> getContributorRoles() {
+		return Common.getList(contributorRoles);
+	}
+
+	/**
+	 * Gets a builder for a {@link Contributor}.
+	 *
+	 * @param contributorType the type of contributor.
+	 * @return the builder.
+	 */
+	public static Builder getBuilder(final ContributorType contributorType) {
+		return new Builder(contributorType);
+	}
+
+	/**
+	 * Gets a builder for a {@link Contributor}.
+	 *
+	 * @param contributorType the type of contributor, as a string.
+	 * @return the builder.
+	 */
+	public static Builder getBuilder(final String contributorType) {
+		return getBuilder(ContributorType.getType(contributorType));
+	}
+
+
+	/** A builder for a {@link Contributor}. */
+	public static class Builder {
+
+		private ContributorType contributorType;
+		private String name = null;
+		private String givenName = null;
+		private String familyName = null;
+		private String contributorID = null;
+		private List<Organization> affiliations = null;
+		private List<ContributorRole> contributorRoles = null;
+
+		private Builder(final ContributorType contributorType) {
+			Objects.requireNonNull(contributorType, "contributor type cannot be null");
+			this.contributorType = contributorType;
+		}
+
+		public Builder withContributorID(final String contributorID) {
+			this.contributorID = Common.checkPid(contributorID, "contributorID", true);
+			return this;
+		}
+
+		public Builder withName(final String name) {
+			this.name = Common.processString(name);
+			return this;
+		}
+
+		public Builder withGivenName(final String givenName) {
+			this.givenName = Common.processString(givenName);
+			return this;
+		}
+		public Builder withFamilyName(final String familyName) {
+			this.familyName = Common.processString(familyName);
+			return this;
+		}
+
+		public Builder withAffiliations(final List<Organization> affiliations) {
+			this.affiliations = Common.processSimpleList(affiliations, "affiliations");
+			return this;
+		}
+
+		public Builder withContributorRoles(final List<ContributorRole> contributorRoles) {
+			this.contributorRoles = Common.processSimpleList(contributorRoles, "contributorRoles");
+			return this;
+		}
+
+		public Builder withContributorRoleStrings(final List<String> contributorStrings) {
```

Comment:
allow `contributorRoles` to be initialised using a list of strings as well as a list of `ContributorRole` objects

Reply:
(`Organization` objects have two fields so it isn't possible to have an analogous `withAffiliationStrings` constructor)